File: 0151-Reverse-Word-in-a-String/0151-Reverse-Word-in-a-String.py
class Solution:
    def reverseWords(self, s: str) -> str:
        words = s.strip().split()
        words.reverse()
        result = ""
        for i in range(len(words)):
            result += words[i]
            if (i < len(words) - 1):
                result += " "
        return result

File: 0151-Reverse-Word-in-a-String/0151-Reverse-Word-in-a-String.java
class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        String result = "";
        String[] words = s.split("\\s+");

        for(int i = words.length - 1; i >= 0; i--){
            result += words[i];
            if(i > 0){
                result += " ";
            }
        }
        return result;
    }
}